analyzers/tests/SonarAnalyzer.UnitTest/AnalysisContext/SonarAnalysisContextTest.Register.cs

```diff
@@ -121,6 +120,47 @@ public void RegisterTreeAction_Extension_SonarParametrizedAnalysisContext_VB()
         context.AssertDelegateInvoked(true);
     }
 
+    [DataTestMethod]
+    [DataRow(SnippetFileName, false)]
+    [DataRow(AnotherFileName, true)]
+    public void RegisterSemanticModelAction_UnchangedFiles_SonarAnalysisContext(string unchangedFileName, bool expected)
+    {
+        var context = new DummyAnalysisContext(TestContext, unchangedFileName);
+        var sut = new SonarAnalysisContext(context, DummyMainDescriptor);
+        sut.RegisterSemanticModelAction(CSharpGeneratedCodeRecognizer.Instance, context.DelegateAction);
+
+        context.AssertDelegateInvoked(expected);
+    }
+
+    [DataTestMethod]
+    [DataRow(SnippetFileName, false)]
+    [DataRow(AnotherFileName, true)]
+    public void RegisterSemanticModelAction_UnchangedFiles_SonarParametrizedAnalysisContext(string unchangedFileName, bool expected)
+    {
+        var context = new DummyAnalysisContext(TestContext, unchangedFileName);
+        var sut = new SonarParametrizedAnalysisContext(new(context, DummyMainDescriptor));
+        sut.RegisterSemanticModelAction(CSharpGeneratedCodeRecognizer.Instance, context.DelegateAction);
+        context.AssertDelegateInvoked(expected);
```

Comment:
Very nitpick: for consistency in the file add an empty line before the assert:
```suggestion
        sut.RegisterSemanticModelAction(CSharpGeneratedCodeRecognizer.Instance, context.DelegateAction);
        
        context.AssertDelegateInvoked(expected);
```